File: tejvanta-ui/src/types/OptionsContract.ts
export interface OptionsContract {
  id: number
  underlyingInstrumentId: number
  strike: number
  expiry: string
  optionType: 'CE' | 'PE'
  ltp: number
  bid: number
  ask: number
  openInterest: number
  volume: number
  iv: number
  change: number
  changePercent: number
}

export interface OptionsChainResponse {
  symbol: string
  expiry: string
  contracts: OptionsContract[]
}

export interface ReplayCandle {
  instrumentId: number
  timestamp: string
  open: number
  high: number
  low: number
  close: number
}

export interface ReplayState {
  instrumentId: number
  isPlaying: boolean
  currentTime: string
  from: string
  to: string
  speed: number
  tickCount: number
  progressPercent: number
}

File: Tejvanta.Api/src/Domain/Entities/OptionsContract.cs
namespace Tejvanta.Api.Domain.Entities;

public class OptionsContract
{
    public int Id { get; set; }
    public int UnderlyingInstrumentId { get; set; }
    public decimal Strike { get; set; }
    public DateTime Expiry { get; set; }
    public string OptionType { get; set; } = "CE";
    public decimal LTP { get; set; }
    public decimal OpenInterest { get; set; }
    public long Volume { get; set; }
    public decimal IV { get; set; }
    public decimal Bid { get; set; }
    public decimal Ask { get; set; }
    public decimal Change { get; set; }
    public decimal ChangePercent { get; set; }
    public Instrument? Underlying { get; set; }
}
